### DISCUSSION TITLE
`apply_force` and `apply_impulse` at position global offset is confusing and misleading

### DISCUSSION THREAD

**Discussion by GustJc** [godotengine/godot-proposals#8044] [3D]
`apply_force` and `apply_impulse` at a position uses a global offset which can be very confusing and misleading.

**The problem:**
I honestly don't think a global offset for a force applied at a point is the most common use case.
I also misunderstood the name `offset` as a local offset(local space position), passing the child object's position where I wanted the forces applied at.
Having to use (`global_position - point_to_apply_node.global_position`) is also very awkward.

**The proposed solution:**
I know we can't change apply_force itself as to not break compatibility, 
But I'd like to propose we add some explicit methods for rigidbody3d like `apply_force_at_world_position` and  `apply_impulse_at_world_position`.

--
As a side note.
Other engines also seem to use a global position when applying forces. With `AddForceAtPosition`(world space) in Unity. And `Add Force at Location Local`(local space) and `Add Force at Location`(world space) in Unreal.

**Comment by rupel190** [comment]
Took me three solid days and precisely this post to figure out how the function works. And while the documentation might technically be correct it didn't help me much using it the first time. 
I presume this is a typical engine-beginner problem and once it's overcome there won't be much incentive about improving given function/docs.

**Reply by Calinou** [reply]
The documentation improvement part is tracked here: https://github.com/godotengine/godot-docs/issues/9066

**Reply by Randommist** [reply]
I literally just walked your path 😢

**Comment by fLindahl** [comment]
Just bumped in to this "problem" as well.

Feels like applying a positional force (or impulse) to an object as an offset to the objects position, but not taking into consideration the rotation is uncommon. The documentation didn't really help either, as "offset from the body origin in global coordinates" can either be understood as the point being in the objects local space (therefore you'd multiply the argument position by the inverse global transform) or the point being in world space (just pass the position). Never in my life would I have read this as "pass a worldspace point, but make subtract the nodes worldspace position first". If the force is in world space, I'd assume the position should be in worldspace, agnostic to where the object is located.

The proposed solution of clarifying the documentation and adding a worldspace function (perhaps deprecating the old) sounds good to me.

**Comment by aarontheo** [comment]
I was about to open an issue as well, I've been running into the same issue. At the very least, we could add a brief, single-line example to the documentation to demonstrate that it must be `body.global_position - point.global_position`. Seems really strange though that it wouldn't just be `point.global_position`, but I don't know how it's been implemented under the hood.

**Comment by GustJc** [comment]
I created a PR a while ago here (https://github.com/godotengine/godot/pull/104900)
It just adds an extra `apply_force_at_position` and `apply_impulse_at_position` that uses the global position.

Since it doesn't change the default of apply_force there is a change this can be merged before 5.0. It would already help I think.
